liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/applications/InstallAppMojoSupport.java

```diff
@@ -216,11 +216,11 @@ private void addLibraryFromM2(Artifact library, LooseConfigData config) throws E
     
     private void validateAppConfig(String fileName) throws Exception {
         String appsDir = getAppsDirectory();
-        if (appsDir.equalsIgnoreCase("apps") && !isFoundAppNameConfigInSourceServerXml(fileName)) {
+        if (appsDir.equalsIgnoreCase("apps") && !isAppConfiguredInSourceServerXml(fileName)) {
             // add application configuration
             applicationXml.createWebApplicationElement(fileName);
         }
-        else if (appsDir.equalsIgnoreCase("dropins") && isFoundAppConfigInSourceServerXml())
+        else if (appsDir.equalsIgnoreCase("dropins") && isAnyAppConfiguredInSourceServerXml())
```

Comment:
Fixed. It should work in either way as It intends to check if any app is configured.